sanic_cors/extension.py

```diff
@@ -387,6 +387,13 @@ def wrapper(cls, f, ctx, req, e):
         else:
             do_await = iscoroutinefunction(f)
             resp = f(req, e)
+            # async Exceptions may be awaitable and should be returned
+            # TODO: iscoroutinefunction may not work as intended for async Exceptions
+            if isawaitable(resp):
```

Comment:
This will always catch all awaitables and coroutines, so the `if do_wait()` under it will never be be reached if true.
You will need to either remove the `do_await` part (combine it with `isawaitable`) or use a different test with equivalent logic.